### PR TITLE
Added Domain Selection Functionality with Validation.

### PR DESCRIPTION
**Pull request by ShivamMenda**
Added a field called in the domain in the sign up page which allows the user to select and search a domain if required in a modal bottom sheet. (Min 3 should be required)

### CODE REVIEW

**Comment by vedantbahel** [comment]
@SirusCodes, Can you help with this code review corresponding to Issue #25 ?

**Comment by Rohit-RA-2020** [comment]
https://user-images.githubusercontent.com/53489549/180832803-a706906b-a28b-4fef-bca7-26480236006f.mp4

### Hey @ShivamMenda, seems like a bug. Can you fix this?

@vedantbahel

**Comment by ShivamMenda** [comment]
@Rohit-RA-2020 Hey I never faced any kind of an issue like this on my device but I'll look into it.

**Review by Rohit-RA-2020 — CHANGES_REQUESTED** [review]
Fix the bug while selecting domains.
Shows min 3 domains even after ticking 3 options.

**Comment by Rohit-RA-2020** [comment]
> @Rohit-RA-2020 Hey I never faced any kind of an issue like this on my device but I'll look into it.

Just a suggestion, make the sheet dismiss after clicking ok only if min 3 options are selected. 
Would be better UX

**Comment by ShivamMenda** [comment]
@Rohit-RA-2020 @vedantbahel I reproduced the same steps and it looks fine to me. 
https://user-images.githubusercontent.com/74780977/180835204-5a0b2bef-670a-45c8-92cc-33a54ac35820.MOV

**Comment by ShivamMenda** [comment]
@vedantbahel @Rohit-RA-2020 I reproduced the same steps and it looks fine to me. https://user-images.githubusercontent.com/74780977/180835622-d2010213-70a3-4ecf-b5e8-9b62c2f4bedf.MOV

**Comment by ShivamMenda** [comment]
Here's the emulator recording.
[recording.webm](https://user-images.githubusercontent.com/74780977/180836488-16754c88-9904-428c-8d59-edc35309e613.webm)

**Comment by ShivamMenda** [comment]
> 
> Just a suggestion, make the sheet dismiss after clicking ok only if min 3 options are selected. Would be better UX

@Rohit-RA-2020 I'll try seeing if this is possible because I don't think the validator can dismiss the sheet, It's just to check based on the values and give error messages after changing state and that's how the package works I guess but I'll try finding a property to dismiss once validated.

File: viteed_app/lib/screens/signup.dart [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 import '../constants/validators.dart';
 
+class Domain {
+  final int id;
+  final String name;
+
+  Domain({
+    required this.id,
+    required this.name,
+  });
```
Comment (SirusCodes) [inline viteed_app/lib/screens/signup.dart:13]: This constructor could be a const constructor

Comment (ShivamMenda) [reply viteed_app/lib/screens/signup.dart:13]: @SirusCodes I'll add a const keyword over here. Thanks.

Comment (SirusCodes) [inline viteed_app/lib/screens/signup.dart:13]: This could be a const constructor

**Review by SirusCodes — CHANGES_REQUESTED** [review]

**Comment by SirusCodes** [comment]
I don't know if all the files in ios folders are required but you can add generated files on `.gitignore`

**Comment by ShivamMenda** [comment]
@SirusCodes Which files should I add to gitignore? All the pod files?

**Comment by SirusCodes** [comment]
That's what I said...idk anything about ios dev😅

**Comment by ShivamMenda** [comment]
@SirusCodes I'm not sure about it either 😂 but as far as I know the pod files that are auto generated I usually don't ignore them because when I ran the app on IOS simulator it automatically updates all internal versions in the files and are important for it to run on IOS. My simulator has IOS 15 so it's better to leave the pod files as it is so that later on it doesn't cause a problem.

**Comment by ShivamMenda** [comment]
@SirusCodes Added const to the constructor.

**Comment by SirusCodes** [comment]
> @SirusCodes I'm not sure about it either 😂 but as far as I know the pod files that are auto generated I usually don't ignore them because when I ran the app on IOS simulator it automatically updates all internal versions in the files and are important for it to run on IOS. My simulator has IOS 15 so it's better to leave the pod files as it is so that later on it doesn't cause a problem. 

Cool then let's hope it's fine then😅

**Comment by Rohit-RA-2020** [comment]
> Here's the emulator recording. [recording.webm](https://user-images.githubusercontent.com/74780977/180836488-16754c88-9904-428c-8d59-edc35309e613.webm) @Rohit-RA-2020

Yes, I also tried re-running the app and it looks good. Don't know what caused the issue but seems good to go👍

**Comment by Rohit-RA-2020** [comment]
> > Just a suggestion, make the sheet dismiss after clicking ok only if min 3 options are selected. Would be better UX
> 
> @Rohit-RA-2020 I'll try seeing if this is possible because I don't think the validator can dismiss the sheet, It's just to check based on the values and give error messages after changing state and that's how the package works I guess but I'll try finding a property to dismiss once validated.

Yeah, try adding this

**Comment by ShivamMenda** [comment]
> Yeah, try adding this

I'll try and let you know @Rohit-RA-2020

**Comment by ShivamMenda** [comment]
@Rohit-RA-2020 So I looked into it but I'm afraid that the package does not have a dismissible property. The User has to press "ok" to update the state and then the validation will occur. I'm sorry I couldn't find a way to dismiss it.

**Comment by vedantbahel** [comment]
Thank you @ShivamMenda for contributing.
Thanks a lot @Rohit-RA-2020 @SirusCodes for the review.

**Comment by ShivamMenda** [comment]
@vedantbahel @Rohit-RA-2020 @SirusCodes Thanks for the PR and Review. Let me know if I can contribute anything else I'll keep track of the project. Great project by the way guys!

**Pull request merged**